### ISSUE TITLE
Allow comments in markdownlint.json

### ISSUE THREAD

**Issue by metablaster** [DavidAnson/vscode-markdownlint#269]
Sample `markdownlint.json`

```json
{
	"_comment": "Disable reporting long lines",
	"MD013": false
}
```

While it's valid json syntax the comment (in quotes, not the keyword) is reported as:
`Incorrect type: expected on of boolean, object`

I propose that markdownlint doesn't report json comments as incorrect.

**Comment by metablaster** [comment]
Yes, in the mean time I've read the docs and learned that markdownlint allows jsonc files so I renamed my configs to jsonc, 
however regardless I think not reporting comments in json is a matter of meticulousness and code correctness.

If you don't agree I'm fine with that since jsonc comments works well.

**Comment by DavidAnson** [comment]
I believe `.markdownlint.json` can have JSONC comments as well. But the editor may get confused about that, so it's easier to use the JSONC extension.
